packages/zosfiles/src/methods/copy/Copy.ts

```diff
@@ -84,4 +92,239 @@ export class Copy {
+        ImperativeExpect.toBeDefinedAndNonBlank(toDataSetName, "toDataSetName");
+
+
+        try {
+            let sourceDataset = options["from-dataset"].dsn;
+            const sourceMember  = options["from-dataset"].member;
+            let   sourceDataSetObj: IZosmfListResponse;
+            let   targetDataset = toDataSetName;
+            const targetMember  = toMemberName;
+            let targetDataSetObj: IZosmfListResponse;
+            let targetFound: boolean = false;
+            let targetMemberFound: boolean = false;
+            let overwriteTarget: boolean = options.replace;
+
+            /**
+             * If no console is passed and no overwrite is passed then set to owerwrite to false
+             */
+            if(console == undefined && overwriteTarget == undefined){
+                overwriteTarget = false;
+            }
+
+            /**
+             * Does the source dataset exist?
+             */
+            const SourceDsList = await List.dataSet(sourceSession, sourceDataset, {
+                attributes: true, maxLength: 1,
+                start: sourceDataset,
+                recall: "wait"
+            });
+
+            if (SourceDsList.apiResponse != null &&
+                SourceDsList.apiResponse.returnedRows != null
+                && SourceDsList.apiResponse.items != null) {
+                // Look for the index of the data set in the response from the List API
+                const dsnameIndex = SourceDsList.apiResponse.returnedRows === 0 ? -1 :
+                    SourceDsList.apiResponse.items.findIndex((ds: any) => ds.dsname.toUpperCase() === sourceDataset.toUpperCase());
+                if (dsnameIndex !== -1) {
+                    // If dsnameIndex === -1, it means we could not find the given data set.
+                    // We will attempt the upload anyways so that we can forward/throw the proper error from z/OS MF
+                    sourceDataSetObj = SourceDsList.apiResponse.items[dsnameIndex];
+                } else {
+                    throw new ImperativeError({ msg: ZosFilesMessages.datasetCopiedAbortedNoTargetDS.message });
+                }
+                /*
+                * If the source is a PDS and no member was specified then abort the copy.
+                */
+                if((sourceDataSetObj.dsorg == "PO" || sourceDataSetObj.dsorg == "POE") && sourceMember == undefined){
+                    throw new ImperativeError({ msg: ZosFilesMessages.datasetCopiedAbortedNoPDS.message });
+                }
+            }
+
+            const task2: ITaskWithStatus = {
+                percentComplete: 0,
+                statusMessage: "Retrieving data set contents",
+                stageName: TaskStage.IN_PROGRESS
+            };
+
+            /**
+             *  Get the dataset and store it in a buffer
+             *
+             *  We want to download and upload in binary, as this command is copying from z/OS to z/OS system.
```

Comment:
We had team conversations about tagging (and loss of tagging) if a binary transfer is specified. Does that concern apply to datasets? If tagging is applicable, would it mean that the target would not have the same tag?

Reply:
I believe tagging to specify encoding only applies to USS files.